UnitsNet.Tests/GeneratedCode/TestsBase/AccelerationTestsBase.g.cs

```diff
@@ -681,322 +681,118 @@ public void TryParse()
+        [InlineData("en-US", "mg", AccelerationUnit.MillistandardGravity)]
+        [InlineData("en-US", "nm/s²", AccelerationUnit.NanometerPerSecondSquared)]
+        [InlineData("en-US", "g", AccelerationUnit.StandardGravity)]
+        [InlineData("ru-RU", "см/с²", AccelerationUnit.CentimeterPerSecondSquared)]
+        [InlineData("ru-RU", "дм/с²", AccelerationUnit.DecimeterPerSecondSquared)]
+        [InlineData("ru-RU", "фут/с²", AccelerationUnit.FootPerSecondSquared)]
+        [InlineData("ru-RU", "дюйм/с²", AccelerationUnit.InchPerSecondSquared)]
+        [InlineData("ru-RU", "км/с²", AccelerationUnit.KilometerPerSecondSquared)]
+        [InlineData("ru-RU", "узел/час", AccelerationUnit.KnotPerHour)]
+        [InlineData("ru-RU", "узел/мин", AccelerationUnit.KnotPerMinute)]
+        [InlineData("ru-RU", "узел/с", AccelerationUnit.KnotPerSecond)]
+        [InlineData("ru-RU", "м/с²", AccelerationUnit.MeterPerSecondSquared)]
+        [InlineData("ru-RU", "мкм/с²", AccelerationUnit.MicrometerPerSecondSquared)]
+        [InlineData("ru-RU", "мм/с²", AccelerationUnit.MillimeterPerSecondSquared)]
+        [InlineData("ru-RU", "мg", AccelerationUnit.MillistandardGravity)]
+        [InlineData("ru-RU", "нм/с²", AccelerationUnit.NanometerPerSecondSquared)]
+        [InlineData("ru-RU", "g", AccelerationUnit.StandardGravity)]
+        public void ParseUnitWithCulture(string culture, string abbreviation, AccelerationUnit expectedUnit)
+        {
+            AccelerationUnit parsedUnit = Acceleration.ParseUnit(abbreviation, CultureInfo.GetCultureInfo(culture));
+            Assert.Equal(expectedUnit, parsedUnit);
+        }
 
-            {
-                Assert.True(Acceleration.TryParseUnit("g", CultureInfo.GetCultureInfo("ru-RU"), out var parsedUnit));
-                Assert.Equal(AccelerationUnit.StandardGravity, parsedUnit);
-            }
+        [Theory]
+        [InlineData("cm/s²", AccelerationUnit.CentimeterPerSecondSquared)]
+        [InlineData("dm/s²", AccelerationUnit.DecimeterPerSecondSquared)]
+        [InlineData("ft/s²", AccelerationUnit.FootPerSecondSquared)]
+        [InlineData("in/s²", AccelerationUnit.InchPerSecondSquared)]
+        [InlineData("km/s²", AccelerationUnit.KilometerPerSecondSquared)]
+        [InlineData("kn/h", AccelerationUnit.KnotPerHour)]
+        [InlineData("kn/min", AccelerationUnit.KnotPerMinute)]
+        [InlineData("kn/s", AccelerationUnit.KnotPerSecond)]
+        [InlineData("m/s²", AccelerationUnit.MeterPerSecondSquared)]
+        [InlineData("µm/s²", AccelerationUnit.MicrometerPerSecondSquared)]
+        [InlineData("mm/s²", AccelerationUnit.MillimeterPerSecondSquared)]
+        [InlineData("mg", AccelerationUnit.MillistandardGravity)]
+        [InlineData("nm/s²", AccelerationUnit.NanometerPerSecondSquared)]
+        [InlineData("g", AccelerationUnit.StandardGravity)]
+        public void TryParseUnit(string abbreviation, AccelerationUnit expectedUnit)
```

Comment:
I pushed some improvements 70f1b3e124e815e5901ad4ede1bb91b19095e262

```
Change tests to control current culture:
- Change `ParseUnit` to `ParseUnit_WithUsEnglishCurrentCulture`
- Change `TryParseUnit` to `TryParseUnit_WithUsEnglishCurrentCulture`

Add tests for more cases:
- ParseUnit_WithUnsupportedCurrentCulture_FallsBackToUsEnglish
- ParseUnit_WithCurrentCulture
- TryParseUnit_WithUnsupportedCurrentCulture_FallsBackToUsEnglish
- TryParseUnit_WithCurrentCulture
```

Reply:
> You can also manipulate the current culture per test, and revert it back, to make the intent more clear and avoid different behavior for different host systems running the tests.

What, like the `ToString` method? Are you sure this is thread safe? I didn't want to risk another #438. I believe NUnit has annotations that can be used to set / ensure a given culture per test, but I don't know how this is supposed to be done properly in xUnit. Especially on a per-test level, I don't know if `try / finally ` is the correct way to do it..

Reply:
XUnit runs each test per test class sequentially, so a try-finally or using-scope pattern to revert changes work well.

However, XUnit will concurrently run test classes by default, unless a test fixture is used to control concurrency. This is not a problem because each test class is run on a different thread, and any changes to `CurrentCulture` or fields marked with `[ThreadStatic]` attribute will be isolated per test class.

The only thing I'm not 100% about is whether XUnit creates new threads per test class or uses som kind of pool that may be tainted, but in my experience this has never been a problem.

Reply:
Oh, that's great- I've been tiptoeing around this issue for a long time.. 
So in what situation was the original issue occurring? There is no way that we could have been configuring something static on the per-test-class level..

Reply:
> So in what situation was the original issue occurring? There is no way that we could have been configuring something static on the per-test-class level..

Not sure what you mean, what issue exactly?
Something about CurrentCulture in tests that failed before?

Generally speaking, since tests can run on any developer or CI agent machine, it's good to control things like `CurrentCulture` to avoid flaky tests.

Reply:
Nah, when I saw the `[Collection(nameof(UnitAbbreviationsCacheFixture))]` (and read on the related issues) I thought that this is something that is required _every time_ we need to change something static..

Reply:
Aha, I didn't remember we had those.

```
// Apply this collection fixture to classes:
        // 1. That rely on manipulating CultureInfo. See https://github.com/angularsen/UnitsNet/issues/436
        // 2. To avoid accessing static ToString/Parse from multiple tests where UnitAbbreviationsCache.Default is modified
```

Nr 1 should not be a problem if you revert the value, not sure what happened back in 2018 but likely the order of test execution caused flakiness if we did not revert values.

Nr 2 is still required, since the static `Default` singleton and other static stuff is not marked `[ThreadStatic]`. If I recall correctly, a test project runs in the same app domain and share static fields, and most of our tests and test classes are in the same project.

Reply:
> Nr 2 is still required, since the static `Default` singleton and other static stuff is not marked `[ThreadStatic]`. If I recall correctly, a test project runs in the same app domain and share static fields, and most of our tests and test classes are in the same project.

Yeah, that's still a bit of a pain in the ass - wish there was a simple annotation that would make a given class run in total isolation, like in BenchmarkDotNet.